Neos.ContentRepository/Classes/Domain/Utility/NodeContexts.php

```diff
@@ -20,12 +20,12 @@
     /**
      * Determine if the given node path is a context path.
      *
-     * @param string $string path or identifier
+     * @param string $string context path or context identifier
      * @return boolean
      */
     public static function hasContext($string)
     {
-        return (strpos($string, '@') !== false);
+        return (preg_match('/@/u', $string) === 1);
```

Comment:
I agree,t he test for the @ is as naive as before but switching to MATCH_PATTERN_CONTEXTPATH and MATCH_PATTERN_CONTEXTIDENTIFIER would actually make the later refactoring of the context more difficult.

In that case we not only would have to check wether those patterns match (they match also without context) an thus we would have to look wether a workspace or dimension was detected by the pattern. That would just be another place of hardcoded context internals.

I altered the change so the `hasContext` method is not needed any more and the `isContextPath` method stays as before but s deprecated